File: backend/src/main/java/com/hireflow/service/JobService.java
package com.hireflow.service;

import com.hireflow.dto.request.JobRequest;
import com.hireflow.dto.response.JobResponse;

import java.util.List;

public interface JobService {

    JobResponse createJob(JobRequest request);

    List<JobResponse> getAllJobs();

    JobResponse getJobById(Long id);

    JobResponse updateJob(Long id, JobRequest request);

    void deleteJob(Long id);

}

File: frontend/src/services/jobService.js
import api from "../api/api";

export const getAllJobs = () => {
    return api.get("/api/jobs");
};

export const getJobById = (id) => {
    return api.get(`/api/jobs/${id}`);
};

export const createJob = (job) => {
    return api.post("/api/jobs", job);
};

export const updateJob = (id, job) => {
    return api.put(`/api/jobs/${id}`, job);
};

export const deleteJob = (id) => {
    return api.delete(`/api/jobs/${id}`);
};